modules/my_preferences/php/my_preferences.class.inc

```diff
@@ -99,21 +99,15 @@ class My_Preferences extends \NDB_Form
      */
     function _process($values)
     {
-        $config = \NDB_Config::singleton();
-        $DB     = \Database::singleton();
+        $factory = \NDB_Factory::singleton();
+        $config  = $factory->config();
+        $DB      = $factory->database();
 
         // build the "real name"
         $values['Real_name'] = $values['First_name'] . ' ' . $values['Last_name'];
 
         $user = \User::factory($this->identifier);
 
-        // If editing a user and nothing was specified in the password text field
-        // remove the key from the value set, otherwise Password_hash
-        // will be set to '' by the system
-        if ($values['Password_hash'] == '') {
-            unset($values['Password_hash']);
-        }
```

Comment:
If the password is not changed, the password won't be in the email. If changed, then yes.

Reply:
These lines were moved below and don't affect what's in the email. Whether the password appears is determined by the `$msg_data` array and the email template.